cmake/scripts/clone_test_data.cmake

```diff
@@ -0,0 +1,16 @@
+# clone test data
+
+get_filename_component(test_data_dir json_test_data ABSOLUTE)
+
+if(NOT EXISTS ${test_data_dir})
+    execute_process(COMMAND ${GIT_EXECUTABLE} clone
+                                -q -c advice.detachedHead=false -b v${JSON_TEST_DATA_VERSION} --depth 1
+				                -- ${JSON_TEST_DATA_URL} ${test_data_dir}
```

Comment:
There're some tab characters here instead of all spaces. I'm only going to fix that if I have to make other changes. I intend to add a `cmake-format` configuration file and reformat all CMake files in the near future (after `clang-format`).
There're many style inconsistencies from spaces before parentheses to how commands are split over multiple lines.